### PR TITLE
Reset driver on depth callback timeout

### PR DESCRIPTION
**Pull request by afakihcpr**

### CODE REVIEW

File: src/astra_driver.cpp [hunk 1]
```diff
@@ -199,6 +203,18 @@ bool AstraDriver::EnableStreaming(std_srvs::SetBool::Request &req, std_srvs::Set
   return true;
 }
 
+void AstraDriver::setHealthTimers() {
+  auto reset_this = [this](const ros::TimerEvent&) -> void
+  {
+    ROS_WARN_STREAM("Astra " << ns_ << " driver timeout! Resetting");
```
Comment (jake5452) [inline src/astra_driver.cpp:213]: This timer callback happens 30s after device_->startDepthStream(); from both line 186 and 544. Given this, shouldn't we check whether or not stream is ok before we decide to destruct and reconstruct the object?

Comment (afakihcpr) [reply src/astra_driver.cpp:213]: If the stream is ok, the depth callback will reset the timer  and the timer callback won't happen at all

File: src/astra_driver.cpp [hunk 2]
```diff
@@ -199,6 +203,18 @@ bool AstraDriver::EnableStreaming(std_srvs::SetBool::Request &req, std_srvs::Set
...
+    auto& nh = nh_;
+    auto& pnh = pnh_;
+    this->~AstraDriver();
+    new (this) AstraDriver(nh, pnh);
+  };
+  depth_callback_timer_ = nh_.createTimer(depth_callback_timeout_, reset_this, false, false);
```
Comment (danielhailuA) [inline src/astra_driver.cpp:219]: `auto reset_this = [this](const ros::TimerEvent&) -> void `
Can reset_this be void?

**Review by jake5452 — APPROVED** [review]
Looks good to me

**Review by danielhailuA — APPROVED** [review]

File: src/astra_driver.cpp (hunk 2 above)
Comment (afakihcpr) [reply src/astra_driver.cpp:219]: the lambda return type is void

**Review by ElizabethHollander — APPROVED** [review]

**Pull request merged**